File: website/app/models/frontend/StateLegislatorWithLatestReportWithNbReports.scala
package models.frontend

import play.api.libs.json.{Json, JsValue, Writes}
import models.{Report, StateLegislator}

case class StateLegislatorWithLatestReportWithNbReports(stateLegislator: StateLegislator,
                                                        latestReport: Option[Report],
                                                        nbReports: Int)

object StateLegislatorWithLatestReportWithNbReports {
  implicit val writes = new Writes[StateLegislatorWithLatestReportWithNbReports] {
    def writes(stateLegislatorWithLatestReportWithNbReports: StateLegislatorWithLatestReportWithNbReports): JsValue = {
      Json.obj(
        "stateLegislator" -> stateLegislatorWithLatestReportWithNbReports.stateLegislator,
        "latestReport" -> stateLegislatorWithLatestReportWithNbReports.latestReport,
        "nbReports" -> stateLegislatorWithLatestReportWithNbReports.nbReports
      )
    }
  }
}

File: website/javascripts/models/stateLegislatorWithLatestReportWithNbReports.js
CBR.Models.StateLegislatorWithLatestReportWithNbReports = new Class({
    Extends: CBR.Models.JsonSerializable,

    options: {  // Defaults
    },

    getStateLegislator: function() {
        return this.options.stateLegislator;
    },

    getLatestReport: function() {
        return this.options.latestReport;
    },

    getNbReports: function() {
        return this.options.nbReports;
    }
});
